test/connectivity/README.md

```diff
@@ -69,3 +72,106 @@ Network offline/online should not affect this test result since Edge hub stores
+##### Test Specific Components:
+	1. Message Sender is a Azure service client that is deployed into a test agent as a container module. The responsibility of the module is to use the Azure API to trigger IoT Hub to send a message to device along with reporting to Test Result Coordinator once a message is sent. 
+	2. Message Receiver is a device client that is wrapped in a module container for a deployment into a test agent. The device module behaves as a target receiving end of the message sending from Message Sender and is responsible for reporting to the Test Result Coordinator once a message is received.
+
+##### Test Implementation Assumption:
+	1. The cloud takes less than 15 minutes to forward all of messages from the cloud to device when the Edge Hub is online.
+	2. Each cycle of Network Controller online-offline period must be longer than the Edge Hub recovering period
+	3. Since the Message Sender uses service client to send a message to the cloud from the host network, it will always report success even if the message has yet gotten to the Message Receiver. If the Message Sender fails to send a message, the message sequencing ID should NOT be increased; instead, a failed message with the same sequence number should be resent.
+	
+##### Test Mechanism: 
+	1. The Message Sender invokes the IoT Hub to send  a message to a targeted device
+	2. The Message Sender notify the Test Result Coordinator that a message has been sent.
+	3. The IoT Hub sends a message to Edge Hub addressing a targeted device
+	4. The Edge Hub forward the message to a Message Receiver device module.
+	5. The Message Receiver notifies that the message successfully received to the Test Result Coordinator
+	6. The Message Receiver replies that the message successfully received to the Edge Hub
+	7. The Edge Hub forward the Message Receiver's response back to IoT Hub
+	8. The IoT Hub forward the response to Message Sender.
+
+##### Passing Criteria: 
+All the messages sent by Message Sender is received in order at Message Receiver within (EdgeHub_Recover_Time) + 15 minutes.
+
+#### EdgeHub Restart Test
+The goal of EdgeHub Restart Test is to verify/benchmark a restart period of an EdgeHub. The measurement is done using two methods: a Direct Method and a Message.
+
+The message test is developed to strictly test only the only when the EdgeHub complete its restart and is ready for a module to connect.  While the direct method test also make sure that the receiver can successfully response to the direct method request from the sender. This results in a slightly different time measurement between the two test (see the diagram below).
+
+![Connectivity Test Diagram](./images/ConnectivityTest_EdgeHubRestart_Timeline.png)
+
+##### EdgeHub Restart Test : Message Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_EdgeHubRestart_Message.png)
+
+###### Test Mechanism: 
```

Comment:
Why Test Components section is missing here? And we should mention EdgeHub Restart Tester and edge agent in host network.

Reply:
OOOH I forgot.